### PR TITLE
Add SHA256 hash to `Program` interface

### PR DESCRIPTION
**Pull request by cblichmann**
Implements the first part of #291

### CODE REVIEW

**Comment by cblichmann** [comment]
Kinda, I haven't actually built the code with my modification yet. I wanted to gauge interest/start a conversation about whether you guys want this.
I'm currently out of office with spotty internet, though. I can still submit I guess.

**Comment by nsadeveloper789** [comment]
It's definitely an improvement, but you're right to ask how valuable it is. In this context, the MD5 was more a quick check to identify the version of a file that was imported when creating the Program. While it would be nice to make this a more secure hash, security was not originally the reason for including it. The contents are recorded in the Program (though loaded rather than the original image), so a user could compare if they're worried about surreptitious changes.

**Comment by cblichmann** [comment]
Right, I did not add this to improve security. It's more useful in the following context: Reverser A does some work and exports thte Ghidra database to some machine. Reverser B takes it, but does not necessarily have the original binary and wants to look up the original sample in, say, Virustotal or another data source. Many malware sources are nowadays easier to search using SHA256 hashes.
This would also definitely avoid MD5 collisions :)

File: Ghidra/Features/Base/src/main/java/ghidra/plugin/importer/ImporterUtilities.java [hunk 1]
```diff
@@ -100,6 +100,10 @@ public static void setProgramProperties(Program program, FSRL fsrl, TaskMonitor
 				fsrl.getMD5() != null) {
 				program.setExecutableMD5(fsrl.getMD5());
 			}
+			if ((program.getExecutableSHA256() == null || program.getExecutableSHA256().isEmpty()) &&
+				fsrl.getSHA256() != null) {
+				program.setExecutableMD5(fsrl.getSHA256());
```
Comment (nsadeveloper789) [inline Ghidra/Features/Base/src/main/java/ghidra/plugin/importer/ImporterUtilities.java]: This and the line above cause a compilation error (getSHA256 does not exist in FSRL). Also, I think you meant to call setExecutableSHA56, not MD5. I'm actually not certain why we try to populate the MD5 in `ImporterUtilties` at all.... That said, I don't think you need to populate the SHA256 here, either. I think your changes to `AbstractProgramLoader` would be sufficient, even when loaded from a special file system. I recommend deleting this change.

Comment (cblichmann) [reply Ghidra/Features/Base/src/main/java/ghidra/plugin/importer/ImporterUtilities.java]: Done (in b799186f980f1374914218f239e4fb81e7bfc941). That's precisely why I labelled it "draft" :)

**Comment by nsadeveloper789** [comment]
Sorry if I was rude. I must confess I got a little too excited, as yours was the first non-trivial contribution. I've put this in for internal review.

**Comment by cblichmann** [comment]
> Sorry if I was rude. I must confess I got a little too excited, as yours was the first non-trivial contribution. I've put this in for internal review.

Haha, no worries. I'm super used to thorough code reviews :)

**Pull request merged**